File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.java
class Solution {
    public int minCostClimbingStairs(int[] cost) {
        int n = cost.length;
        int a = cost[n - 1];
        int b = cost[n - 2];
        for(int i = n - 3; i >= 0 ; i--){
            int temp = b;
            b = Math.min(a , b) + cost[i];
            a = temp;
        }
        return Math.min(a,b);
    }
}

File: 0746-min-cost-climbing-stairs/0746-min-cost-climbing-stairs.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        if(len(cost) <= 2):
            return min(cost[1], cost[0])
        lc = len(cost)
        two = cost[lc - 1]
        one = cost[lc - 2]
        for i in range(lc - 3, -1, -1):
            temp = min(two, one) + cost[i]
            two = one
            one = temp
        return min(one, two)
        
        